src/main/java/model/Movable.java

```diff
@@ -1,32 +1,123 @@
 package model;
 
+import model.directions.Direction;
+
 public class Movable extends Entity {
-	private final Direction direction;
+	private final int speed;
+	private Direction direction;
+	protected boolean hasToMove = false;
 
-	public Movable(Position center, int majorRadius, int minorRadius, Direction direction) {
+	public Movable(Position center, int majorRadius, int minorRadius, int speed, Direction direction) {
 		super(center, majorRadius, minorRadius);
+		this.speed = speed;
 		this.direction = direction;
 	}
 
-	protected void move(Direction direction) {
+	protected void collision() {
 	}
 
-	protected void collision() {
+	public boolean getHasToMove() {
+		return hasToMove;
+	}
+
+	public void setHasToMove(boolean hasToMove) {
+		this.hasToMove = hasToMove;
 	}
 
 	@Override
-	protected int getXRadius() {
-		if (direction.equals(Direction.RIGHT) || direction.equals(Direction.LEFT)) {
+	public int getXRadius() {
```

Comment:
I don't like this.. u ask for the getXRadius and if some condition happends u return the YRadius? that doesn't make sense..
What did u try to do?

Reply:
Esto tampoco, está modificado.